lxd/device/disk.go

```diff
@@ -1012,3 +1047,65 @@ func (d *disk) getParentBlocks(path string) ([]string, error) {
 
 	return devices, nil
 }
+
+// generateVMConfigDrive generates an ISO containing the cloud init config for a VM.
+// Returns the path to the ISO.
+func (d *disk) generateVMConfigDrive() (string, error) {
+	configDrivePath := filepath.Join(d.instance.Path(), "configdrv")
+
+	// Create config drive dir.
+	err := os.MkdirAll(configDrivePath, 0100)
+	if err != nil {
+		return "", err
+	}
+
+	// Add config drive mount instructions to cloud init.
+	vendorData := `#cloud-config
```

Comment:
@stgraber should this vendor data be read from `instanceConfig["user.vendor-data"]` like the others?

Reply:
@stgraber fixed